File: tests/org/elixir_lang/reference/callable/Issue429Test.java
package org.elixir_lang.reference.callable;

import com.intellij.psi.PsiElement;
import com.intellij.psi.search.LocalSearchScope;
import com.intellij.psi.search.SearchScope;
import com.intellij.testFramework.fixtures.BasePlatformTestCase;
import org.elixir_lang.psi.call.Call;

public class Issue429Test extends BasePlatformTestCase {
    /*
     * Tests
     */

    public void testUseScope() {
        myFixture.configureByFiles("get_use_scope.ex");
        PsiElement callable = myFixture
                .getFile()
                .findElementAt(myFixture.getCaretOffset())
                .getPrevSibling()
                .getLastChild()
                .getLastChild()
                .getLastChild()
                .getLastChild()
                .getLastChild();

        assertInstanceOf(callable, Call.class);
        SearchScope useScope = callable.getUseScope();

        assertInstanceOf(useScope, LocalSearchScope.class);

        LocalSearchScope localSearchScope = (LocalSearchScope) useScope;
        PsiElement[] scope = localSearchScope.getScope();

        assertEquals(1, scope.length);
        PsiElement singleScope = scope[0];

        assertTrue(
                "Use Scope is not the surrounding if",
                singleScope.getText().startsWith("if auth == ")
        );
    }

    /*
     * Protected Instance Methods
     */

    @Override
    protected String getTestDataPath() {
        return "testData/org/elixir_lang/reference/callable/issue_429";
    }
}

File: tests/org/elixir_lang/reference/callable/Issue429Test.kt
package org.elixir_lang.reference.callable

import com.intellij.psi.search.LocalSearchScope
import org.elixir_lang.PlatformTestCase
import org.elixir_lang.psi.call.Call

class Issue429Test : PlatformTestCase() {
    fun testUseScope() {
        myFixture.configureByFiles("get_use_scope.ex")
        val callable = myFixture
                .file
                .findElementAt(myFixture.caretOffset)!!
                .prevSibling
                .lastChild
                .lastChild
                .lastChild
                .lastChild
                .lastChild
        assertInstanceOf(callable, Call::class.java)

        val useScope = callable.useScope
        assertInstanceOf(useScope, LocalSearchScope::class.java)
        val localSearchScope = useScope as LocalSearchScope

        val scope = localSearchScope.scope
        assertEquals(1, scope.size)

        val singleScope = scope[0]
        assertTrue(
                "Use Scope is not the surrounding if",
                singleScope.text.startsWith("if auth == ")
        )
    }

    override fun getTestDataPath(): String = "testData/org/elixir_lang/reference/callable/issue_429"
}
